### PR TITLE
Request state service

### PR DESCRIPTION
**Pull request by igorT**
[Rendered](https://github.com/igorT/rfcs/blob/igor/request-state-service/text/0466-request-state-service.md)

### CODE REVIEW

File: text/0000-requeest-state-service.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+Expose a service on the store
+
+```
+class Store {
+  getRequestStateService(): RequestStateService
+}   
+```
+
+Using these  we can reimplement the current `isSaving` method on `DS.Model`
```
Comment (mehulkar) [inline text/0000-requeest-state-service.md]: Is this saying we can specifically re-implement isSaving, or does it cover isLoading, et. al? 

Pretty much every ember app I’ve ever worked on has computed properties that have the isLoaded flag as a dependent key.

Comment (igorT) [reply text/0000-requeest-state-service.md]: You will be able to reimplement all of the flags on the model that are based on the request state.

**Comment by igorT** [comment]
Action item: discuss unsubscribe method @runspired @rwjblue

**Comment by igorT** [comment]
Figure out where to export `unsubscribe` from

**Comment by fullsailor** [comment]
Minor nit: The filename has a typo. `s/requeest/request/`

**Comment by igorT** [comment]
Open question, that shouldn't impact the FCP period, figure out the import path for the unsubscribe method

File: text/0000-requeest-state-service.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+
+// exported from '@ember-data/store';
+function unsubscribe(token: UnsubscribeToken): void
+
+class RequestStateService {
+  getPendingRequests(recordIdentifier: RecordIdentifier): RequestState[]
+  getLastRequest(recordIdentifier: RecordIdentifier): RequestState | null
+  subscribe(recordIdentifier: RecordIdentifier, callback: Function): UnsubscribeToken
```
Comment (hjdivad) [inline text/0000-requeest-state-service.md]: todo: clarify callback args

**Comment by igorT** [comment]
Consider renaming 
```
getPendingRequests
getLastRequest
subscribe
```
to
```
getPendingRequestsForRecord
getLastRequestForRecord
subscribeForRecord
```

**Comment by dgeb** [comment]
Discussed this further with @igorT today and agreed to the following: 
* the record-specific methods on the `RequestStateService` should have record-specific names (e.g. `getPendingRequestsForRecord`).
* the `RequestState` interface `type` can be potentially expanded in the future to allow for hybrid requests.
* the decision on whether methods that take a `RecordIdentifier` on `RequestStateService` should also optionally take a `Record` comes down to a matter of consistency (which is probably best decided by @igorT, @runspired, and others who are deeper into the implementation than I am)

I have no objections to merging now.

File: text/0000-requeest-state-service.md (hunk 2 above)
Comment (igorT) [reply text/0000-requeest-state-service.md]: clarified

**Comment by igorT** [comment]
Blocked merging on @runspired's proposal for changing the interfaces

**Comment by dgeb** [comment]
I've reviewed the latest changes and approve.

**Pull request merged**